src/derivation.test.ts

```diff
@@ -235,61 +249,36 @@ describe('derivation', () => {
         await expect(
           bip32Derive({
             path: input,
-            privateKey: Buffer.alloc(32).fill(1),
-            chainCode: Buffer.alloc(32).fill(1),
+            node,
           }),
         ).rejects.toThrow(
           'Invalid BIP-32 index: The index must be a non-negative decimal integer less than 2147483648.',
         );
       }
+    });
 
+    it('throws when no node is specified', async () => {
       await expect(
         bip32Derive({
-          path: `44`,
-          chainCode: Buffer.alloc(32).fill(1),
+          path: '0',
         }),
       ).rejects.toThrow(
-        'Invalid parameters: Must specify either a parent private or public key.',
+        'Invalid parameters: Must specify a node to derive from.',
       );
     });
 
-    it('throws for an invalid private key', async () => {
-      await expect(
-        bip32Derive({
-          path: `44'`,
-          privateKey: Buffer.alloc(31).fill(1),
-          chainCode: Buffer.alloc(32).fill(1),
-        }),
-      ).rejects.toThrow('Invalid parent key: Must be 32 bytes long.');
-    });
-
-    it('throws for an invalid public key', async () => {
-      await expect(
-        bip32Derive({
-          path: `44`,
-          publicKey: Buffer.alloc(31).fill(1),
-          chainCode: Buffer.alloc(32).fill(1),
-        }),
-      ).rejects.toThrow('Invalid parent public key: Must be 65 bytes long.');
-    });
-
-    it('throws if no chain code is specified', async () => {
-      await expect(
-        bip32Derive({
-          path: `44'`,
-          privateKey: Buffer.alloc(32, 1),
-        }),
-      ).rejects.toThrow('Invalid parameters: Must specify a chain code.');
-    });
+    it('throws when trying to derive from a public key node', async () => {
+      const node = await bip39Derive({ path: mnemonic });
+      const publicNode = node.neuter();
 
-    it('throws for an invalid chain code', async () => {
```

Comment:
How come we don't test the chain code validation anymore?

Reply:
The function doesn't accept a plain chain code anymore, and `SLIP10Node` validates any inputs upon creating the node.